### ISSUE TITLE
Generate method provider bottleneck

### ISSUE THREAD

**Issue by dantleech** [phpactor/phpactor#1634]
The generate method provider takes almost 20 seconds in the Phpactor `ReflectionClassTest`
```
Diagnostic finsihed in "18.98" (Phpactor\Extension\LanguageServerCodeTransform\CodeAction\GenerateMethodProvider)
```

**Comment by dantleech** [comment]
this really needs tobe solved by running the node context resolver "once" and providing diagnostics in a single pass and ideally in a separate worker process.

**Comment by dantleech** [comment]
Fixed in #1673
